### PR TITLE
Condition refactor

### PR DESCRIPTION
**Pull request by bcsaller**
This branch is a bit of a monster. 

It includes changes in rules/model that change the engine runtime
It also contains an associated UI refactor that helped me debug some of these issues (and has slightly nicer UX)

### CODE REVIEW

File: tests/test_prog [hunk 1]
```diff
@@ -11,10 +11,10 @@
       "periodic": 5
       "do":
         "action": matrix.plugins.actions.health
-      "until": glitch.complete
+      "until": chaos.complete
     - "after": deployment.healthy
       "do":
-        "action": matrix.plugins.actions.glitch
+        "action": tests.chaos.chaos
```
Comment (petevg) [inline tests/test_prog]: Is this an intentional reversion to the old naming scheme?

Comment (bcsaller) [inline tests/test_prog]: no, this was me needing tests that don't require redeploy after each run

**Comment by johnsca** [comment]
The new UI looks great.  However, when I tried to run it, it got stuck on health running.  Even with reverting the glitch / chaos action that Pete mentioned, glitch never went from pending to running.

**Comment by petevg** [comment]
Other than one potential issue w/ a chaos -> glitch name getting reverted, this looks good to me. +1

**Pull request merged**